src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -268,9 +268,7 @@ private async Task ProcessHeadersFrame(FrameHeader frameHeader)
             {
                 _incomingBuffer.Discard(frameHeader.Length);
 
-                // Don't wait for completion, which could happen asynchronously.
-                ValueTask ignored = SendRstStreamAsync(streamId, Http2ProtocolErrorCode.StreamClosed);
-                return;
+                throw new Http2ProtocolException(Http2ProtocolErrorCode.ProtocolError);
```

Comment:
I'll actually make this update now. This should definitely be an error of type `Http2ProtocolErrorCode.StreamClosed` instead of the general `Http2ProtocolErrorCode.ProtocolError`. Thanks.

Reply:
Ok.  Thanks.  Why is this instance still `.ProtocolError` ?

Reply:
Because I incorrectly updated the wrong instance. Fixed in the latest commit.